connector/spanmetricsconnector/internal/metrics/metrics.go

```diff
@@ -211,22 +211,40 @@ func (h *explicitHistogram) Observe(value float64) {
 	h.bucketCounts[index]++
 }
 
-func (h *explicitHistogram) AddExemplar(traceID pcommon.TraceID, spanID pcommon.SpanID, value float64) {
-	e := h.exemplars.AppendEmpty()
-	e.SetTraceID(traceID)
-	e.SetSpanID(spanID)
-	e.SetDoubleValue(value)
+func (h *explicitHistogram) AddExemplar(traceID pcommon.TraceID, spanID pcommon.SpanID, value float64, maxCount int) {
```

Comment:
Assuming we make `maxCount` type `*int`, the implementation could be simplified to:
```go
	if maxCount != nil && h.exemplars.Len() >= *maxCount {
		return
	}
	e := s.exemplars.AppendEmpty()
	e.SetTraceID(traceID)
	e.SetSpanID(spanID)
	e.SetDoubleValue(value)
```